community/ndp/kernelextension/src/main/java/org/neo4j/ext/NDPKernelExtension.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.ext;
+
+import java.util.concurrent.TimeUnit;
+
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.config.Setting;
+import org.neo4j.graphdb.factory.Description;
+import org.neo4j.helpers.HostnamePort;
+import org.neo4j.helpers.Service;
+import org.neo4j.kernel.GraphDatabaseAPI;
+import org.neo4j.kernel.configuration.Config;
+import org.neo4j.kernel.extension.KernelExtensionFactory;
+import org.neo4j.kernel.impl.util.JobScheduler;
+import org.neo4j.kernel.impl.util.StringLogger;
+import org.neo4j.kernel.lifecycle.LifeSupport;
+import org.neo4j.kernel.lifecycle.Lifecycle;
+import org.neo4j.kernel.logging.Logging;
+import org.neo4j.ndp.transport.http.HttpTransport;
+import org.neo4j.ndp.transport.http.SessionRegistry;
+import org.neo4j.ndp.runtime.NDPRuntime;
+import org.neo4j.ndp.runtime.internal.SingleThreadedNDPRuntime;
+
+import static org.neo4j.helpers.Settings.BOOLEAN;
+import static org.neo4j.helpers.Settings.DURATION;
+import static org.neo4j.helpers.Settings.HOSTNAME_PORT;
+import static org.neo4j.helpers.Settings.setting;
+import static org.neo4j.kernel.impl.util.JobScheduler.Group.serverTransactionTimeout;
+
+/**
+ * Wraps Mjolnir and exposes it as a Kernel Extension.
+ */
+@Service.Implementation(KernelExtensionFactory.class)
+public class NDPKernelExtension extends KernelExtensionFactory<NDPKernelExtension.Dependencies>
```

Comment:
Two: 1) I'd rather do that integration incrementally, moving the code in first and plugging things in one step at a time and 2) we've said we won't plug this in yet, so I'm staying with that plan until @akollegger tells me otherwise